typings/index.d.ts

```diff
@@ -2118,7 +2118,7 @@ declare module 'discord.js' {
     public edit(command: ApplicationCommandResolvable, data: ApplicationCommandData): Promise<ApplicationCommand>;
     public fetch(id: Snowflake, options?: BaseFetchOptions): Promise<ApplicationCommand>;
     public fetch(id?: Snowflake, options?: BaseFetchOptions): Promise<Collection<Snowflake, ApplicationCommand>>;
-    public fetchPermissions(command: undefined): Promise<Collection<Snowflake, ApplicationCommandPermissions[]>>;
+    public fetchPermissions(command?: null): Promise<Collection<Snowflake, ApplicationCommandPermissions[]>>;
     public fetchPermissions(command: ApplicationCommandResolvable): Promise<ApplicationCommandPermissions[]>;
```

Comment:
```suggestion
    public fetchPermissions(): Promise<Collection<Snowflake, ApplicationCommandPermissions[]>>;
    public fetchPermissions(command: ApplicationCommandResolvable): Promise<ApplicationCommandPermissions[]>;
```

Reply:
not sure if this will allow passing null values but then again cant really think of a reason it should

Reply:
Why should it allow passing null values?

Reply:
it was suggested by @vaporox but i believe passing null should not mean "fetch me all command perms instead"

Reply:
The argument is just listed as optional on the JSDoc of the function implementation, the type isn't `ApplicationCommandResolvable|null`

https://github.com/discordjs/discord.js/blob/02693bc02f45980d8165820a103220f0027b96b7/src/managers/ApplicationCommandManager.js#L212

Reply:
Completely removing the parameter here would also make sense, however for the `ApplicationCommandManager#fetchPermissions()` overload at https://github.com/discordjs/discord.js/blob/master/typings/index.d.ts#L2078-L2081 (which needs to be updated too), removing it is not possible because it is followed by another parameter, so I thought it would make sense to stay consistent between the two methods.

Reply:
Perhaps making the function take an "options" parameter would be a better approach than the current one

Reply:
Yeah, it also comes from another context since it concerns the application command manager rather than the guild based one such as my updated overload, so there is no need for a null ApplicationCommandResolvable scenario